File: easy/swap_elements.rb
File.open(ARGV[0]).each_line do |line|
  s = line.chomp.split
  swap = false
  r = []
  s.each do |i|
    if swap
      u, v = i.chomp(',').split('-').map(&:to_i)
      r[u], r[v] = r[v], r[u]
    elsif i == ':'
      swap = true
    else
      r << i
    end
  end
  puts r.join(' ')
end

File: easy/swap_elements.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp $line;
    my $swap = 0;
    my @r;
    foreach ( split( / /, $line ) ) {
        if ($swap) {
            $_ =~ /(\d+)-(\d+)/;
            ( $r[$1], $r[$2] ) = ( $r[$2], $r[$1] );
        }
        elsif ( $_ eq ":" ) {
            $swap = 1;
        }
        else {
            push @r, $_;
        }
    }
    printf( "%s\n", join( ' ', @r ) );
}
close(INFILE);
